generator/src/main/java/com/scottlogic/deg/generator/generation/FieldSpecValueGeneratorFactory.java

```diff
@@ -0,0 +1,41 @@
+package com.scottlogic.deg.generator.generation;
+
+import com.scottlogic.deg.generator.Field;
+import com.scottlogic.deg.generator.fieldspecs.FieldSpec;
+import com.scottlogic.deg.generator.generation.databags.DataBag;
+import com.scottlogic.deg.generator.generation.databags.DataBagSource;
+
+import java.util.stream.Stream;
+
+public class FieldSpecValueGeneratorFactory {
+    private final FieldValueSourceEvaluator evaluator;
+
+    public FieldSpecValueGeneratorFactory(FieldValueSourceEvaluator evaluator) {
+        this.evaluator = evaluator;
+    }
+
+    public DataBagSource getFieldSpecValueGenerator(Field field, FieldSpec fieldSpec){
+        return new FieldSpecValueGeneratorInverter(field, fieldSpec, evaluator);
+    }
+
+    class FieldSpecValueGeneratorInverter implements DataBagSource {
```

Comment:
This type exists to bridge the gap between where we want to be (`Field` and `FieldSpec` are not dependency types) and where they are dependencies.

This type inverts function arguments into dependencies, and vice-versa.